tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -200,6 +203,34 @@ private static void AgentConvertSpan<TInner>(TInner activity, Span span)
             {
                 span.Type = activity5 is null ? SpanTypes.Custom : AgentSpanKind2Type(activity5.Kind, span);
             }
+
+            // extract any ActivityLinks
+            ExtractActivityLinks<TInner>(span, activity5);
+        }
+
+        private static void ExtractActivityLinks<TInner>(Span span, IActivity5? activity5)
+            where TInner : IActivity
+        {
+            if (activity5 is null)
+            {
+                return;
+            }
+
+            var links = new List<IActivityLink>();
+            foreach (var link in (activity5.Links))
+            {
+                var duckLink = link.DuckCast<IActivityLink>();
+                links.Add(duckLink!);
+            }
+
+            if (links.Count <= 0)
+            {
+                return;
+            }
+
+            var settings = new JsonSerializerSettings { Converters = new List<JsonConverter> { new ActivityLinkConverter() }, Formatting = Formatting.None };
```

Comment:
Nice and clean! 👍🏼

Reply:
As a result of this PR, we no longer need to serialize into JSON. Instead, we need to convert the ActivityLink into our own SpanLink representation, then the logic in `tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs` will take care of serializing the SpanLink into Message Pack -- NOTE: We may need to update that file in this PR